File: api/src/main/java/de/hipp/pnp/api/dto/LanguageRequest.java
package de.hipp.pnp.api.dto;

import org.springframework.stereotype.Component;

@Component
public class LanguageRequest {
    String locale;
    Integer gameType;

    public LanguageRequest(Integer gameType, String locale) {
        this.gameType = gameType;
        this.locale = locale;
    }

    public LanguageRequest() {
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }

    public Integer getGameType() {
        return gameType;
    }

    public void setGameType(Integer gameType) {
        this.gameType = gameType;
    }
}

File: api/src/main/kotlin/de/hipp/pnp/api/dto/LanguageRequest.kt
package de.hipp.pnp.api.dto

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.Pattern
import org.springframework.stereotype.Component

/**
 * Language request DTO with input validation.
 *
 * @property locale Locale code in format xx_XX or xx-XX (e.g., en_US, de_DE)
 * @property gameType Game type identifier (0-100)
 */
@Component
data class LanguageRequest(
    @field:Pattern(
        regexp = "^[a-z]{2}[_-][A-Z]{2}$",
        message = "Locale must be in format xx_XX or xx-XX (e.g., en_US, de_DE)"
    )
    var locale: String? = null,

    @field:Min(0, message = "Game type must be non-negative")
    @field:Max(100, message = "Game type must not exceed 100")
    var gameType: Int = 0
)
